src/third-party/EnsController.ts

```diff
@@ -0,0 +1,136 @@
+import BaseController, { BaseConfig, BaseState } from '../BaseController';
+import { isValidAddress, toChecksumAddress } from 'ethereumjs-util';
+
+/**
+ * @type EnsEntry
+ *
+ * ENS entry representation
+ *
+ * @property chainId - Id of the associated chain
+ * @property ensName - The ENS name
+ * @property address - Hex address with the ENS name
+ */
+export interface EnsEntry {
+	chainId: string;
+	ensName: string;
+	address: string;
+}
+
+/**
+ * @type EnsState
+ *
+ * ENS controller state
+ *
+ * @property ensEntries - Object of ENS entry objects
+ */
+export interface EnsState extends BaseState {
+	ensEntries: { [chainId: string]: { [ensName: string]: EnsEntry } };
+}
+
+/**
+ * @type EnsEntriesState
+ *
+ * ENS controller state
+ *
+ * @property ensEntries - Object of ENS entry objects
+ */
+interface EnsEntriesState {
```

Comment:
why no extends BaseState?

Reply:
That interface referred to a substate, but it turned out to be unnecessary so I removed it.  `EnsState` does extend `BaseState`.